### PR TITLE
Add new limit: max_series_label_size_bytes

### PR DESCRIPTION
**Pull request by harry671003**
Signed-off-by: 🌲 Harry 🌊 John 🏔 <johrry@amazon.com>

**What this PR does**:
This PR introduces a new limit `max_series_label_size_bytes`.
Currently, it is possible for users of Cortex to OOM kill distributors and ingesters by increasing the size of each label and label value. For instance, a user can send 1 million series with each series having 30 labels of maximum size. `1 million * 30 labels * 3 KB per label = 85 GB`.

This new limit can prevent this from happening by rejecting series that have a combined label size greater than this limit.
Example: if the `max_series_label_size_bytes` is set to 10 KB, any series that exceed will be rejected. 1 million active series for a user will consume `1 million * 10 KB = 9 GB`.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [X] Tests updated
- [X] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/util/validation/validate.go [hunk 1]
```diff
@@ -216,6 +230,23 @@ func ValidateLabels(cfg LabelValidationConfig, userID string, ls []cortexpb.Labe
...
+
+	maxSeriesLabelSizeBytesMutex.Lock()
+	defer maxSeriesLabelSizeBytesMutex.Unlock()
+	if prev, ok := maxSeriesLabelSizeBytesMap[userID]; ok {
+		if prev > maxSeriesLabelSizeBytes {
+			maxSeriesLabelSizeBytesMap[userID] = maxSeriesLabelSizeBytes
+			MaxSeriesLabelSizeBytesPerUser.WithLabelValues(userID).Set(float64(maxSeriesLabelSizeBytes))
+		}
+	} else {
+		maxSeriesLabelSizeBytesMap[userID] = maxSeriesLabelSizeBytes
```
Comment (alanprot) [inline pkg/util/validation/validate.go:249]: Im worried about this lock here!

Could we do this not on the hot path? Maybe use a defer or even do it on compactor?

File: docs/configuration/config-file-reference.md [hunk 2]
```diff
@@ -2553,6 +2553,11 @@ The `limits_config` configures default and per-tenant limits imposed by Cortex s
 # CLI flag: -validation.max-label-names-per-series
 [max_label_names_per_series: <int> | default = 30]
 
+# Maximum combined size in bytes of all labels and label values accepted for a
+# series. 0 to disable the limit.
+# CLI flag: -validation.max-series-label-size-bytes
+[max_series_label_size_bytes: <int> | default = 0]
```
Comment (alvinlin123) [inline docs/configuration/config-file-reference.md:2559]: I think we should be in consistent naming with other label limits and have the config name to be `max_labels_size_bytes`

File: pkg/util/validation/validate.go (hunk 1 above)
Comment (alvinlin123) [reply pkg/util/validation/validate.go:249]: What would be the use case for these metrics?

Comment (harry671003) [reply pkg/util/validation/validate.go:249]: Before we can enable this, we need to make sure existing users are not impacted. This metrics provide a way to monitor the current largest series in terms of total bytes that each user have.

File: docs/configuration/config-file-reference.md (hunk 2 above)
Comment (harry671003) [reply docs/configuration/config-file-reference.md:2559]: Renamed the flat to `max_labels_size_bytes`

File: pkg/util/validation/validate.go (hunk 1 above)
Comment (harry671003) [reply pkg/util/validation/validate.go]: I've removed this metric for now. The plan is to add it later in the private repo and not upstream.

**Review by alanprot — APPROVED** [review]

**Review by alvinlin123 — APPROVED** [review]

**Comment by alvinlin123** [comment]
Thanks for the contribution @harry671003 !

**Pull request merged**